src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -67,6 +67,9 @@ internal static bool CanHaveAssociatedLocalBinder(this CSharpSyntaxNode syntax)
                 kind == SyntaxKind.CatchFilterClause ||
                 kind == SyntaxKind.SwitchSection ||
                 kind == SyntaxKind.MatchSection ||
+                kind == SyntaxKind.EqualsValueClause ||
```

Comment:
This looks a bit weird given the comment above explicitly excluding `FieldInitializerBinders` but `EqualsValueClause` being present in field initializers. Perhaps we need a new description in the comment?